src/v2/view-builder/views/DeviceChallengePollView.js

```diff
@@ -140,12 +141,23 @@ const Body = BaseForm.extend(Object.assign(
       };
 
       const doProbing = () => {
-        return checkPort()
+        this.checkPortXhr = checkPort();
+        return this.checkPortXhr
           // TODO: can we use standard ES6 promise methods, then/catch?
-          .done(onPortFound)
+          .done(() => {
+            this.probingXhr = onPortFound();
+            return this.probingXhr;
+          })
           .fail(onFailure);
       };
 
+      this.listenTo(this.options.appState, 'invokeAction', (action) => {
```

Comment:
Use `handleInvokeAction` method of `FormController` instead of intercepting it here.

Reply:
`handleInvokeAction` is in the controller, not in the view. This change is only needed in this view. How do you suggest we handle it?

Reply:
All the form actions, We try to handle in FormController irrespective of which view is this.. I also see that `this.listenTo(this.options.appState, 'invokeAction')` invokeAction is not listened by any other view or used anywhere else but FormController..

In my opinion actions should be handled by FormController only.. Try to debug `handleInvokeAction` method and see if you can generalize `cancel` polling action.. You can pass the context of your view to handler to get the methods that you want to abort..